### PR TITLE
feat: update phishing controller

### PR DESCRIPTION
**Pull request by tommasini**
## **Description**

Updating phishing controller to v^5.0.0, this PR is part of controller update v53.0.0 

## **Related issues**

Fixes: #

## **Screenshots/Recordings**
https://recordit.co/VQcpDUfAgn

### **E2E**
https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/c33e72ac-ab6c-4456-989a-0ea05726ddbb -> smoke pipeline
https://app.bitrise.io/build/15b62033-dbc6-4d23-86bc-467605801aaf -> metamask android qa to check launch times.

image
image
image

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/store/migrations.js [hunk 1]
```diff
@@ -773,6 +773,26 @@ export const migrations = {
...
   },
+  /**
+   * This migration is to free space of unused data in the user devices
+   * regarding the phishing list property listState, that is no longer used
+   * @param {any} state - Redux state
+   * @returns
+   */
+  26: (state) => {
+    if (state?.engine?.backgroundState?.PhishingController?.listState) {
+      delete state.engine.backgroundState.PhishingController.listState;
```
Comment (MarioAslau) [inline app/store/migrations.js:784]: PR looks great!

If I recall in other migrations we would sometimes copy the data first then delete the old shape.

Would it be possible to double check when moving from an old version that the migration happens without any issues?

**Review by MarioAslau — APPROVED** [review]
LGTM

File: app/store/migrations.js (hunk 1 above)
Comment (tommasini) [reply app/store/migrations.js:784]: When initializing the engine we update the new list if needed, and we do not need the values right away, I do not think it's needed to copy the old list to the new format, but will ask a second opinion!

Great catch!

**Review by Cal-L — APPROVED** [review]
LGTM

**Pull request merged**